lib/puppet/provider/archive/ruby.rb

```diff
@@ -219,6 +219,53 @@ def s3_download(path)
     aws(params)
   end
 
+  def check_latest
+    # Check if the file exists, just download it if not
+    if File.exist?(archive_filepath)
+      localmodifytime = File.mtime(archive_filepath).to_i
+      case resource[:source]
+      when %r{^(http)}
+        uri = URI(resource[:source])
+        response = nil
+        Net::HTTP.start(uri.host, uri.port,
```

Comment:
I've just used the function that is available in PuppetX::Bodeco::Util. This has all the needed checks and functions already. No need for me to re implement what is already there.